packages/language-server/src/completion-items.ts

```diff
@@ -27,41 +27,75 @@ import {
 } from "@ui5-language-assistant/xml-views-completion";
 import { ui5NodeToFQN } from "@ui5-language-assistant/logic-utils";
 import { getNodeDocumentation, getNodeDetail } from "./documentation";
+import { getSettingsForDocument } from "@ui5-language-assistant/settings";
 import { assertNever } from "assert-never";
 
-export function getCompletionItems(
-  model: UI5SemanticModel,
-  textDocumentPosition: TextDocumentPositionParams,
-  document: TextDocument
-): CompletionItem[] {
-  const documentText = document.getText();
+export async function getCompletionItems(opts: {
+  model: UI5SemanticModel;
+  textDocumentPosition: TextDocumentPositionParams;
+  document: TextDocument;
+}): Promise<CompletionItem[]> {
+  const documentText = opts.document.getText();
   const { cst, tokenVector } = parse(documentText);
   const ast = buildAst(cst as DocumentCstNode, tokenVector);
   const suggestions = getXMLViewCompletions({
-    model: model,
-    offset: document.offsetAt(textDocumentPosition.position),
+    model: opts.model,
+    offset: opts.document.offsetAt(opts.textDocumentPosition.position),
     cst: cst as DocumentCstNode,
     ast: ast,
     tokenVector: tokenVector,
   });
 
+  const filteredSuggestions = await filterBySettings(
+    opts.document.uri,
+    suggestions
+  );
+
   const completionItems = transformToLspSuggestions(
-    suggestions,
-    model,
-    textDocumentPosition.position
+    filteredSuggestions,
+    opts.model,
+    opts.textDocumentPosition
   );
   return completionItems;
 }
 
+async function filterBySettings(
```

Comment:
I believe the actual filtering should happen in the xml-view-completion package.
It is not the responsibility of the language-server package, it is **semantics** of UI5 not
relevant for the LSP response transformations.

Reply:
I don't disagree with this, I just wasn't sure if we want to do it in that level because it might be in several places and we'll need to pass the document URI. Though if we do it in `getXMLViewCompletions` it would only be in one place which is reasonable. Should I move it to there?

Reply:
Changed